### PR TITLE
fix(expressive-modal): ensure waiting for transitionend event

### PR DESCRIPTION
**Pull request by asudoh**
### Related Ticket(s)

Refs #5173.

### Description

Changes the "focus on the primary focusable element" logic in `<dds-expressive-modal>` to wait for `transitionend` event. Without it, the attempt to focus fails.

Also changes `<dds-locale-modal>` to make `<dds-region-item>` as the primary focusable element.

### Changelog

**Changed**

- The "focus on the primary focusable element" logic in `<dds-expressive-modal>` to wait for `transitionend` event.
- `<dds-locale-modal>` to make `<dds-region-item>` as the primary focusable element.

### CODE REVIEW

File: packages/web-components/src/components/locale-modal/locale-modal.ts [hunk 1]
```diff
@@ -187,6 +187,16 @@ class DDSLocaleModal extends DDSExpressiveModal {
     return `${ddsPrefix}-locale-search`;
   }
 
+  /**
+   * A selector selecting the nodes that should be focused when modal gets open.
+   */
+  static get selectorPrimaryFocus() {
+    return `
+      [data-modal-primary-focus],
+      ${ddsPrefix}-region-item
```
Comment (kennylam) [inline packages/web-components/src/components/locale-modal/locale-modal.ts:196]: Thanks for this @asudoh! Can we target the close button instead to be consistent with all our other modals?

```suggestion
      ${ddsPrefix}-expressive-modal-close-button
```

**Review by kennylam — COMMENTED** [review]
Thanks for jumping in on this @asudoh! Can you target the release branch please `release/v1.16.0`?

**Comment by asudoh** [comment]
> Can you target the release branch please `release/v1.16.0`?

Done - Thank you for pointing it out @kennylam!

**Review by kennylam — APPROVED** [review]
LGTM! Thanks @asudoh!

**Pull request merged**